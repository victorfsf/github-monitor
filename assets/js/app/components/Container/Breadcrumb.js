import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './styles.scss';


const Breadcrumb = props => (
  <nav className="breadcrumb" aria-label="breadcrumb">
    <div className="container">
      {props.items && [{
        name: 'Home',
        link: '/',
        active: props.home,
      }, ...props.items].map((i) => {
        if (i.active) {
          return (
            <span className="breadcrumb-item active" aria-current="page" key={i.name}>
              {i.name}
            </span>
          );
        }
        return (
          <span className="breadcrumb-item" key={i.name}>
            <Link to={i.link}>{i.name}</Link>
          </span>
        );
      })}
    </div>
  </nav>
);


Breadcrumb.propTypes = {
  home: PropTypes.bool,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      name: PropTypes.string,
      link: PropTypes.link,
    }),
  ),
};


Breadcrumb.defaultProps = {
  items: [],
  home: false,
};


export default Breadcrumb;
