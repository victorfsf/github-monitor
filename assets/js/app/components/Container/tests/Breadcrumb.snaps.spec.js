import React from 'react';
import renderer from 'react-test-renderer';
import Breadcrumb from 'app/components/Container/Breadcrumb';
import * as ReactRouter from 'react-router-dom';

ReactRouter.Link = jest.fn(props => (
  <a href={props.to}>{props.children}</a>
));

describe('Breadcrumb', () => {
  let Component;
  let tree;

  test('Renders without items', () => {
    Component = renderer.create(<Breadcrumb />);
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Renders with items', () => {
    const items = [{
      active: false,
      name: 'Commits',
      link: '/commits/',
    }, {
      active: true,
      name: 'Repository',
      link: '/repo/',
    }];
    Component = renderer.create(<Breadcrumb items={items} />);
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
