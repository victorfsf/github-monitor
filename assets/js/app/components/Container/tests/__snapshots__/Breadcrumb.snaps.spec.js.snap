// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Renders without items 1`] = `
<nav
  aria-label="breadcrumb"
  className="breadcrumb"
>
  <div
    className="container"
  >
    <span
      className="breadcrumb-item"
    >
      <a
        href="/"
      >
        Home
      </a>
    </span>
  </div>
</nav>
`;

exports[`Breadcrumb Renders with items 1`] = `
<nav
  aria-label="breadcrumb"
  className="breadcrumb"
>
  <div
    className="container"
  >
    <span
      className="breadcrumb-item"
    >
      <a
        href="/"
      >
        Home
      </a>
    </span>
    <span
      className="breadcrumb-item"
    >
      <a
        href="/commits/"
      >
        Commits
      </a>
    </span>
    <span
      aria-current="page"
      className="breadcrumb-item active"
    >
      Repository
    </span>
  </div>
</nav>
`;
